reactive/kotlinx-coroutines-reactor/src/Flux.kt

```diff
@@ -32,9 +35,24 @@ import kotlin.coroutines.*
  * **Note: This is an experimental api.** Behaviour of publishers that work as children in a parent scope with respect
  *        to cancellation and error handling may change in the future.
  */
+
 @ExperimentalCoroutinesApi
 fun <T> CoroutineScope.flux(
     context: CoroutineContext = EmptyCoroutineContext,
     @BuilderInference block: suspend ProducerScope<T>.() -> Unit
-): Flux<T> =
-    Flux.from(publish(newCoroutineContext(context), block = block))
+): Flux<T> = Flux.from(reactorPublish(newCoroutineContext(context), block = block))
+
+
+private fun <T> CoroutineScope.reactorPublish(
+    context: CoroutineContext = EmptyCoroutineContext,
+    @BuilderInference block: suspend ProducerScope<T>.() -> Unit
+): Publisher<T> = Publisher { subscriber ->
+    // specification requires NPE on null subscriber
+    if (subscriber == null) throw NullPointerException("Subscriber cannot be null")
+    val currentContext = if (subscriber is CoreSubscriber) subscriber.currentContext() else Context.empty()
```

Comment:
I've asked Sebastian to verify whether this approach is stable or not.
As a backup plan, `Flux.create` can be used by adapting `FluxSink` to `Subscriber` and using corresponding callbacks: https://www.screencast.com/t/2Z2fFBq6HSC

Reply:
Reactor team responded that the current way is enough, let's stick to it.
But I'd rather throw an exception if a subscriber is not `CoreSubscriber`